src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java

```diff
@@ -25,7 +24,12 @@ public User add(User user) {
     }
 
     @Override
-    public Optional<User> findByEmail(String email) {
-        return userDao.findByEmail(email);
+    public User findByEmail(String email) {
+        return userDao.findByEmail(email).get();
```

Comment:
```suggestion
        return userDao.findByEmail(email).orElse(null);
```

I would recommend to do this
and replace `isPresent()` check with null check in AuthenticationServiceImpl